### PR TITLE
scorep/30rc1: gnu build fails in build-backend/libscorep_measurement.la

### PR DESCRIPTION
**Pull request by jgphpc**
@geimer In preparation for CSCS upgrade, i am trying to build scorep/30rc1 on Cray/CLE6
using CrayGNU/2016.06. The recipe fails with:

```
  libtool:   error: error: cannot install 'libscorep_measurement.la' to a directory 
not ending in /apps/common/UES/sandbox/jgp/efface/easybuild/software/
OTF2/2.0/lib/backend
  Makefile:13796: recipe for target 'install-libLTLIBRARIES' failed
  make[4]: *** [install-libLTLIBRARIES] Error 1
  make[4]: Leaving directory '/dev/shm/piccinal/ScoreP/3.0-rc1/
CrayGNU-2016.06/scorep-3.0-rc1/build-backend'
```

My manual workaround is to manually edit build-backend/libscorep_measurement.la and to replace:

```
libdir='/home/users/p01991/EB/easybuild/software/OTF2/2.0/lib/backend'
with:
libdir='path_to_install_dir/lib/backend'
```
- Do you know how to do this from the easyconfig file ?
- Is there a fix for this issue ?

```
brisi01: libtool (GNU libtool) 2.4.2
```

### CODE REVIEW

**Comment by boegel** [comment]
@jgphpc how does that change fix anything? isn't `path_to_install_dir=/home/users/p01991/EB/easybuild/software/OTF2/2.0`?

At which step of the installation process to do you need to make this change? After configure/before build?

**Comment by jgphpc** [comment]
### How does that change fix anything ?

> libdir='/home/users/p01991/EB/easybuild/software/OTF2/2.0/lib/backend'
> with:
> libdir='path_to_install_dir/lib/backend'

@boegel _path_to_install_dir_ is (for example):

`/apps/common/UES/sandbox/jgp/efface/easybuild/software/Score-P/3.0-rc1-CrayGNU-2016.06/lib/backend 
`
My point was that libdir incorrectly points to OTF2. 

Compiling OTF2 internally with score-p (instead of precompiling OTF2) seems to fix my issue with libtool (see my latest commit), i'll check what happens with other compilers than gnu.

### At which step of the installation process to do you need to make this change ? After configure/before build ?
After the configure but during the build in build-backend (multiple dirs are compiled), maybe @geimer can give more details ?

**Comment by boegel** [comment]
@jgphpc I'm not sure why that happens, looks like a bug to me, but you should be able to fix it with something like

```
prebuildopts = "sed -i 's@libdir=.*@libdir=%(installdir)s/lib/backend@g' build-backend/libscorep_measurement.la && "
```

**Comment by geimer** [comment]
@jgphpc This patching shouldn't be necessary.  I'm not yet sure why you are seeing this error.  We have never seen it on any platform, not even on another Cray system where we installed Score-P recently (though w/o EasyBuild).  Is there a way I can leverage your EasyBuild environment on daint, for example?

File: easybuild/easyconfigs/s/Score-P/Score-P-3.0-rc1-CrayGNU-2016.06.eb [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+configopts += ' --with-libpmi-include=/opt/cray/pe/pmi/default/include'
+configopts += ' --with-libpmi-lib=/opt/cray/pe/pmi/default/lib64'
+configopts += ' --with-librca-include=/opt/cray/rca/default/include'
+configopts += ' --with-librca-lib=/opt/cray/rca/default/lib64'
+# -------
+# gcc: error: unrecognized command line option '-craype-verbose'
+# because -craype-verbose is supported only by the Cray wrappers (cc):
+# => fix it by setting CFLAGS/CXXFLAGS
+configopts += ' CC=cc CFLAGS=-O2'       # !needed for build-score
+configopts += ' CXX=CC CXXFLAGS=-O2'    # !needed for build-score
```
Comment (geimer) [inline easybuild/easyconfigs/s/Score-P/Score-P-3.0-rc1-CrayGNU-2016.06.eb]: This workaround is no longer necessary; fixed in Score-P 2.0.2 and above.

File: easybuild/easyconfigs/s/Score-P/Score-P-3.0-rc1-CrayGNU-2016.06.eb [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+sanity_check_paths = {
+    'files': ["bin/scorep", "include/scorep/SCOREP_User.h",
+              ("lib/backend/libscorep_adapter_mpi_event.a",
+              "lib/libscorep_adapter_mpi_event.%s" % SHLIB_EXT) ],
```
Comment (geimer) [inline easybuild/easyconfigs/s/Score-P/Score-P-3.0-rc1-CrayGNU-2016.06.eb]: Should be
```
    ("lib/backend/libscorep_adapter_mpi_event.a", "lib64/backend/libscorep_adapter_mpi_event.a"),
    ("lib/backend/libscorep_adapter_mpi_event.%s" % SHLIB_EXT,
     "lib64/backend/libscorep_adapter_mpi_event.%s" % SHLIB_EXT)],
```

**Comment by jgphpc** [comment]
This pull request can be closed if you prefer.

**Comment by geimer** [comment]
@jgphpc Does that mean that the latest commits resolved the issues you encountered?

**Comment by boegel** [comment]
@jgphpc can you come back to @geimer's question? Also: please close this PR yourself (we try to avoid closing PRs by someone else).

**Comment by jgphpc** [comment]
> Does that mean that the latest commits resolved the issues you encountered?
Yes, compiling OTF2 "internally" fixes my issue.

**Comment by geimer** [comment]
@jgphpc I'm still interested in understanding why it doesn't work with an external OTF2 installation.  It should -- and if it doesn't, it's a bug...

**Comment by geimer** [comment]
@jgphpc Confirmed.  I was able to reproduce your issue.  It is likely to be a bug in the Score-P build system.  We will investigate how to resolve this.

**Comment by migueldiascosta** [comment]
@jgphpc, this PR is being closed for the following reason(s): no activity for > 6 months.
Please don't hesitate to reopen this PR or add a comment if you feel this contribution is still relevant.
For more information on our policy w.r.t. closing PRs, see https://easybuild.readthedocs.io/en/latest/Contributing.html#why-a-pull-request-may-be-closed-by-a-maintainer

**Pull request closed without merging**